README.md

```diff
@@ -1,25 +1,106 @@
- Systemd integration for the Warp web server (WAI).
- 
- # Features
+# warp-systemd
+
+[Systemd](https://systemd.io/) integration for the [Warp Web Server (WAI)](https://github.com/yesodweb/wai).
+
+## Features
 
 - [Socket Activation](http://0pointer.de/blog/projects/socket-activation.html)
 - [Watchdog](http://0pointer.de/blog/projects/watchdog.html)
 
-# Usage
+## Usage
 
-By default all systemd integrations are off. Tweak `SystemdSettings` to enable watchdog for example:
+All systemd integrations are off by default. For example, you can modify `SystemdSettings` to enable the watchdog:
 
 ```haskell
+{-# LANGUAGE OverloadedStrings #-}
+
 import Data.Function                                 ((&))
 import qualified Network.Wai.Handler.Warp         as Warp
 import qualified Network.Wai.Handler.Warp.Systemd as Systemd
+import Network.Wai
+import Network.HTTP.Types.Status                     (status200)
 
-myWaiApp = ...
+app :: Application
+app _ respond =
+  respond $ responseLBS status200 [] "Hello World"
 
 main :: IO ()
-main = do
-  Systemd.runSystemdWarp systemdSettings Warp.defaultSettings myWaiApp
+main =
+  Systemd.runSystemdWarp systemdSettings Warp.defaultSettings app
   where
-    systemdSettings = Systemd.SystemdSettings
-      & setHeartbeatInterval (Just 3)
-```
\ No newline at end of file
+    systemdSettings = Systemd.defaultSystemdSettings
+      & Systemd.setHeartbeatInterval (Just 5)
+```
+
+## Systemd Service Example
+
+Let's configure your app to run as a socket-activated systemd service. We'll be using the sample app in [example/Main.hs](example/Main.hs).
+
+1. Create a new service unit at `/etc/systemd/system/warp.service` that'll run your Haskell binary:
+
+   ```systemd
+   [Unit]
+   Description=Warp
+   [Service]
+   Type=notify
+   WatchdogSec=5
```

Comment:
I usually add margin, or does systemd decide that? Otherwise having the same number of seconds seems like cutting it really close.

Reply:
Good point. Maybe 30 seconds?

Reply:
Sgtm